serialzy/serializers/tensorflow.py

```diff
@@ -0,0 +1,118 @@
+import inspect
+import logging
+import os
+import shutil
+import tempfile
+from abc import ABC
+from typing import Union, Type, Callable, Dict, Any, BinaryIO, Optional
+
+from packaging import version
+
+from serialzy.api import Schema, VersionBoundary
+from serialzy.base import DefaultSchemaSerializerByReference
+from serialzy.utils import cached_installed_packages
+
+_LOG = logging.getLogger(__name__)
+
+
+def serialize_dir(dir: str, dest: BinaryIO):
```

Comment:
Maybe it is reasonable to generalize these functions and move them to utils? It is very likely that they will be useful for other serializers